### ISSUE TITLE
Hydrate Manager from json

### ISSUE THREAD

**Issue by chrismichaels84** [chrismichaels84/data-manager#15]
It should be possible to hydrate an instance from json. Maybe one of:

```php
$manager->fromJson($json);
$manager->resetFromJson($json);
$manager->initFromJson($json);
```

**Comment by smolinari** [comment]
In reference to our discussion [here](https://github.com/spider/spider/issues/64#issuecomment-139841930), I'd like to take a crack at this.

Anything you'd like me to look out for?

Scott

**Comment by chrismichaels84** [comment]
That's great! I would really appreciate it. Just be sure to branch off of `develop`. It should be relatively simple, please be sure to write a test or two and update the readme for it as well. If you have any questions, let me know. Make the changes to `ManagesItemsTrait`.

**Comment by smolinari** [comment]
Ok. I have a first attempt working. 

However, `json_decode()` can decode to an object or an array and I currently have the assoc array option set to true. Do we only want an array created from `json_decode`? Or do we want to give the option to create an object too?

Would there be a possible request to hydrate from other formats (XML, YAML)? Being the docs say the manager would be good for managing config data, I thought this might be useful or needed later, so I built in a parameter for the type. 

Scott

**Comment by chrismichaels84** [comment]
That's great. Can you issue a PR so I can see the code and test the build?

I think `json_decode` should decode to an array so you can just use one of the internal methods to add like `reset()`. And, yes, it would be great to hydrate from other formats. I had planned on starting with json, then yaml, and then allowing some kind of `parser` system so users can add their own formats for the rest of the craziness out there.

It also just occurred to me that there are possibly two things users may want. 
  1. The initial hydration from json or rehydration from json (which would wipe all current data)
  2. And *adding* json data to the already existing stack.

Maybe two different methods? Luckily, Manager already has `reset()` and `add()` which should help.

**Comment by smolinari** [comment]
I am not certain about the big picture for sure, but I find it hard to imagine a hydration from external sources as an appending type of function being common. I'd say, make it a reset and leave a possible appending feature for the future, should someone ask for it. 

Scott

**Comment by smolinari** [comment]
Actually, never mind. I can think of a scenario. I'll create it. 

`hydrateFrom('json', $data)` causes a reset and creates a fresh manager.
`appendFrom('json', $data)` causes an adding of the data to the current manager.

Scott

**Comment by smolinari** [comment]
Ok. PR is up!

https://github.com/chrismichaels84/data-manager/pull/26

Scott

**Comment by smolinari** [comment]
Edit: Hold on a sec...

Scott

**Comment by smolinari** [comment]
Ok. That looks better. I think. LOL! :smiley: 

Scott

**Comment by chrismichaels84** [comment]
Merged #26.
